src/Diagnostics/FxCop/CSharp/Design/CodeFixes/CA1052CSharpCodeFixProvider.cs

```diff
@@ -39,16 +36,37 @@ public sealed override FixAllProvider GetFixAllProvider()
             if (classDeclaration != null)
             {
                 var title = string.Format(AnalyzerPowerPackRulesResources.StaticHolderTypeIsNotStatic, classDeclaration.Identifier.Text);
-                var codeAction = new MyCodeAction(title, ct => AddStaticKeyword(document, root, classDeclaration));
+                var codeAction = new MyCodeAction(title, ct => MakeClassStatic(document, root, classDeclaration));
                 context.RegisterCodeFix(codeAction, context.Diagnostics);
             }
         }
 
-        private Task<Document> AddStaticKeyword(Document document, SyntaxNode root, ClassDeclarationSyntax classDeclaration)
+        private Task<Document> MakeClassStatic(Document document, SyntaxNode root, ClassDeclarationSyntax classDeclaration)
         {
-            var staticKeyword = SyntaxFactory.Token(SyntaxKind.StaticKeyword).WithAdditionalAnnotations(Formatter.Annotation);
-            var newDeclaration = classDeclaration.AddModifiers(staticKeyword);
-            var newRoot = root.ReplaceNode(classDeclaration, newDeclaration);
+
+            SyntaxTokenList modifiers = classDeclaration.Modifiers;
```

Comment:
It might be simpler and less error prone to use DocumentEditor here. Something like:

```csharp
var editor = await DocumentEditor.CreateAsync(document);
editor.SetModifiers(classDeclaration, editor.GetModifiers(classDeclaration) - DeclarationModifiers.Sealed + DeclarationModifiers.Static);
```